src/ml-search.service.js

```diff
@@ -1135,11 +1135,55 @@
       var self = this;
       
       // TODO: what other conditions
+      // [GJo] This looks a bit peculiar. If you have start and limit in params, but no options, then they get lost?
       if ( !options && params && params.options) {
         options = params;
         params = null;
       }
```

Comment:
Sounds safer, I'll add.

This usage style is not cover in the comment above the function. But I see no nice way to add it there..